FinSearchUnified/Bundle/SearchBundleFindologic/SortingHandler/PopularitySortingHandler.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace FinSearchUnified\Bundle\SearchBundleFindologic\SortingHandler;
+
+use FinSearchUnified\Bundle\SearchBundleFindologic\FindologicSortingHandlerInterface;
+use FinSearchUnified\Bundle\SearchBundleFindologic\QueryBuilder;
+use Shopware\Bundle\SearchBundle\Sorting\PopularitySorting;
+use Shopware\Bundle\SearchBundle\SortingInterface;
+use Shopware\Bundle\StoreFrontBundle\Struct\ShopContextInterface;
+
+class PopularitySortingHandler implements FindologicSortingHandlerInterface
+{
+    /**
+     * Checks if the passed sorting can be handled by this class
+     *
+     * @param SortingInterface $sorting
+     *
+     * @return bool
+     */
+    public function supportsSorting(SortingInterface $sorting)
+    {
+        return $sorting instanceof PopularitySorting;
+    }
+
+    /**
+     * Handles the passed sorting object.
+     *
+     * @param SortingInterface $sorting
+     * @param QueryBuilder $query
+     * @param ShopContextInterface $context
+     */
+    public function generateSorting(SortingInterface $sorting, QueryBuilder $query, ShopContextInterface $context)
+    {
+        /** @var PopularitySorting $sorting */
```

Comment:
Why is this necessary? Shouldn't the IDE already know about the accessible methods (since it implements the interface)?

Please have a second look and remove the line from each sorting handler if it isn't needed.

Reply:
Please check the comment for the same in condition handler PR

Reply:
Ah, okay. It would work without it if we used the abstract sorting class instead of the interface. Anyhow, it can stay the way it is.